### ISSUE TITLE
wercker-init failing when setting different Docker host

### ISSUE THREAD

**Issue by soudy** [wercker/wercker#255]
I'm trying to set a different docker host using this command:
```bash
wercker build --pipeline build --docker-local --tag=local --docker-host=tcp://192.168.99.100:2376 --docker-tls-verify="1" --docker-cert-path="/home/soud/.minikube/certs
```

However, it fails on wercker-init. Seems like the file isn't being created:
```
--> Running step: wercker-init
/bin/bash: line 54: /pipeline/wercker-init-a122bf50-2de6-40dc-83f0-d6bf7b319e7b/init.sh: No such file or directory
--> Step failed: wercker-init 0.02s
--> Pipeline failed: 1.91s
FATAL Step failed: wercker-init
```

Running build without custom docker configs works just fine.

Verbose info from a later run:
```
INFO[0011]  literal| [x]  stdin "source \"/pipeline/wercker-init-98376f9b-ea43-4fcf-ab38-d2257e6d0578/init.sh\"\n" 
INFO[0011]  literal| [x] stdout "/bin/bash: line 54: /pipeline/wercker-init-98376f9b-ea43-4fcf-ab38-d2257e6d0578/init.sh: No such file or directory\n" 
``` 

Any clue what's causing this?

**Comment by hatchan** [comment]
We require that the host that is running the docker daemon has the source directory on the same location as where wercker is running. 

With Linux this isn't an issue since both are are the same. 

Docker for mac (and docker-machine) mounts the user directory from the host to the same location inside the VM. So it works fine (as long as your source code is in your home directory).

I noticed you are using minikube, you might have to check if minikube is doing the same trick as Docker for Mac. Maybe it is configurable, or you might need to manually add it.

**Comment by soudy** [comment]
Thanks @hatchan, this is on Ubuntu and I just tested it on an OSX machine and it worked fine. Seems to be a problem with mounting host directories in boot2docker or minikube, not sure yet. I'll investigate further and report back for people walking into the same issue.

**Comment by hatchan** [comment]
Looks like support was added for the `/Users` directory: https://github.com/kubernetes/minikube/pull/434

I remember `/users/` being tricky for boot2docker/docker-machine, since there already is a `/users/` directory on the image.

**Comment by soudy** [comment]
Yes, xhyve support was added, virtualbox not yet: https://github.com/kubernetes/minikube/issues/716. There are currently 2 PRs open that implement it and they're looking to merge one. In the meantime I'll mount over NFS as a temporary workaround for Linux. So this is not a wercker issue but a minikube one.

**Comment by b-long** [comment]
Any update on this issue?

**Comment by ConradIrwin** [comment]
Is there a workaround if I don't have everything in my home directory?

**Comment by iuriaranda** [comment]
I'm having the same problem if I run wercker from the new Bash for Windows console. Wercker itself runs just fine, but of course Docker for windows requires windows paths (i.e. `C:\Users\...`) to mount volumes in the containers, and wercker provides the Linux path.

Not sure if there's a solution to that, but as there's no wercker cli for windows, I had to try this 😄
